user/tests/wiring/no_fixture_power_saving/power_saving_mode.cpp

```diff
@@ -0,0 +1,235 @@
+        }
+        // Log.info("period: %lu", period);
+
+    } while (!lowPowerResult && attempts++ < LOW_POWER_ATTEMPTS_MAX);
+    assertMoreOrEqual(period, 1280 - 128); // 1.28s +/- 10% if in low power mode
+    assertLessOrEqual(period, 1280 + 128);
+}
+
+test(POWER_SAVING_03_call_function_and_check_return_value_after_low_power_active) {
+    if (modem_type != DEV_SARA_R510) {
+        skip();
+        return;
+    }
+
+    // See no_fixture_long_running.spec.js
+}
+
+test(POWER_SAVING_04_check_function_argument_value) {
+    if (modem_type != DEV_SARA_R510) {
+        skip();
+        return;
+    }
+
+    // Loop a bit before we check this to make sure the device app thread has received this message.
+    // 5s is kind of long, but we are not validating how fast it can receive this data, just that it
+    // does.  We want this test to be reliable.
+    uint32_t start = millis();
+    while (millis() - start < 5000) {
+        Particle.process();   // pump application events
+    }
+    assertTrue(fnLp1Arg == "argument string low power");
+}
+
+test(POWER_SAVING_05_system_sleep_with_configuration_object_ultra_low_power_mode_wake_by_network) {
+    if (modem_type != DEV_SARA_R510) {
+        skip();
+        return;
+    }
+
+    Particle.connect();
+    assertTrue(waitFor(Particle.connected, CLOUD_CONNECT_TIMEOUT));
+
+    delay(15s); // a bit of delay required to avoid premature wake, even with SystemSleepFlag::WAIT_CLOUD
+    bool lowPowerResult = false;
+    int attempts = 0;
+    uint32_t period = 0;
+    const uint32_t WAIT_FOR_LOW_POWER_ACTIVE_MS = 20000;
+    const uint32_t WAIT_FOR_LOW_POWER_MEAS_MS = 10000;
+    const uint32_t LOW_POWER_ATTEMPTS_MAX = 10; // XXX: extreme cases take up to 2.5 minutes to drop into low power mode
+    do {
+        Particle.connect();
+        assertTrue(waitFor(Particle.connected, CLOUD_CONNECT_TIMEOUT));
+        delay(WAIT_FOR_LOW_POWER_ACTIVE_MS); // wait for UPSV=1 default delay of ~9.2s before modem drops into low power mode idle mode.
+
+        period = measureAvgPeriodMs(CTS1, WAIT_FOR_LOW_POWER_MEAS_MS);
+        if (period <= 1280 + 128 && period >= 1280 - 128) { // 1.28s +/- 10% if in low power mode
+            lowPowerResult = true;
+        } else {
+            lowPowerResult = false;
+        }
+        // Log.info("period: %lu", period);
+
+    } while (!lowPowerResult && attempts++ < LOW_POWER_ATTEMPTS_MAX);
+    assertMoreOrEqual(period, 1280 - 128); // 1.28s +/- 10% if in low power mode
+    assertLessOrEqual(period, 1280 + 128);
+
+    assertEqual(0, pushMailbox(MailboxEntry().type(MailboxEntry::Type::RESET_PENDING), 30000));
+    system_tick_t start = millis();
+    SystemSleepConfiguration config;
+    config.mode(SystemSleepMode::ULTRA_LOW_POWER)
+        .duration(60s)
+        .network(NETWORK_INTERFACE_CELLULAR);
+    SystemSleepResult result = System.sleep(config);
+
+    // in sleep for 60s, should wake up after 20s due to function call
```

Comment:
The function call should come from the test-runners `before()` function right? 
It looks like 30s after every test start the cloud side will call the test function on the device, which should wake it in this isntance

Reply:
Right, it was 20s and I increased it to 30s.

Reply:
Updated the comment.